adapters/repos/db/lsmkv/segment_serialization_benchmarks_test.go

```diff
@@ -63,3 +71,310 @@ func BenchmarkCollectionNodeKeyIndexAndWriteTo(b *testing.B) {
+	b.SetBytes(int64(len(data)))
+	b.ReportAllocs()
+
+	for i := 0; i < b.N; i++ {
+		reader.Reset(fileContents)
+		err = ParseReplaceNodeIntoPread(reader, uint16(tc.secondaryKeysCount), out)
+		if err != nil {
+			b.Fatal("error parsing test data:", err)
+		}
+	}
+}
+
+// runFileBufferingOnly measures the cost of reading the file into a memory buffer (without parsing).
+// This isolates the I/O overhead from parsing.
+func runFileBufferingOnly(b *testing.B, tc struct {
+	name               string
+	valueSize          int
+	keySize            int
+	secondaryKeysCount int
+	secondaryKeySize   int
+}, data []byte, tempFile string) {
+	file, cleanup := openFile(b, tempFile)
+	defer cleanup()
+
+	b.ResetTimer()
+	b.SetBytes(int64(len(data)))
+	b.ReportAllocs()
+
+	for i := 0; i < b.N; i++ {
+		if _, err := file.Seek(0, 0); err != nil {
+			b.Fatal("Failed to seek file:", err)
+		}
+
+		all, err := io.ReadAll(file) // Includes measuring the cost of reading the file into memory.
+		if err != nil {
+			b.Fatal("error reading data:", err)
+		}
+		runtime.KeepAlive(all) // Prevents compiler optimizations.
+	}
+}
+
+// runPreloadBufferAccess measures parsing performance when the file has already been loaded into memory and a shared buffer is used.
+// This isolates the parsing cost and avoids measuring I/O overhead.
+func runPreloadBufferAccess(b *testing.B, tc struct {
+	name               string
+	valueSize          int
+	keySize            int
+	secondaryKeysCount int
+	secondaryKeySize   int
+}, data []byte, tempFile string, out *segmentReplaceNode) {
+	fileContents, err := os.ReadFile(tempFile) // File read before benchmark timing.
+	if err != nil {
+		b.Fatal("Failed to read file:", err)
+	}
+
+	reader := bytes.NewReader(fileContents)
+	b.ResetTimer()
+	b.SetBytes(int64(len(data)))
+	b.ReportAllocs()
+
+	for i := 0; i < b.N; i++ {
+		reader.Reset(fileContents) // Resets reader instead of allocating a new one.
+		err = ParseReplaceNodeIntoPread(reader, uint16(tc.secondaryKeysCount), out)
+		if err != nil {
+			b.Fatal("error parsing test data:", err)
+		}
+	}
+}
+
+// runBufferedFileAccess measures performance of a buffered approach where the entire file is read into memory first,
+// then parsed from a memory buffer. Includes the cost of reading the file into memory.
+func runBufferedFileAccess(b *testing.B, tc struct {
+	name               string
+	valueSize          int
+	keySize            int
+	secondaryKeysCount int
+	secondaryKeySize   int
+}, data []byte, tempFile string, out *segmentReplaceNode) {
+	file, cleanup := openFile(b, tempFile)
+	defer cleanup()
+
+	b.ResetTimer()
+	b.SetBytes(int64(len(data)))
+	b.ReportAllocs()
+
+	for i := 0; i < b.N; i++ {
+		if _, err := file.Seek(0, 0); err != nil {
+			b.Fatalf("Failed to seek file: %v", err)
+		}
+
+		all, err := io.ReadAll(file) // Reading full file into memory.
+		if err != nil {
+			b.Fatal("error reading file:", err)
+		}
+
+		if len(all) == 0 {
+			b.Fatalf("file is empty")
+		}
```

Comment:
```suggestion
		require.NotEmptyf(b, all /* `content`? */, "file %q is empty", file.Name())
```

Another example for the suggestion above ⬆️ -- [NotEmpty](https://github.com/stretchr/testify/blob/v1.10.0/require/require.go#L1478)